src/ducks/loan/LoanDetailsPage.spec.jsx

```diff
@@ -0,0 +1,212 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { mount } from 'enzyme'
+import {
+  CreditReserveSection,
+  CharacteristicsSection,
+  PaymentsSection,
+  KeyInfosSection,
+  Section,
+  Row
+} from './LoanDetailsPage'
+
+describe('keys infos section', () => {
+  describe('when at least one relevant data is defined', () => {
+    it('should render defined data', () => {
+      const wrapper = mount(
+        <KeyInfosSection
+          account={{
+            loan: {
+              usedAmount: 10000,
+              rate: 2
+            },
+            balance: -2000
+          }}
+        />,
+        {
+          context: {
+            t: key => key
+          },
+          childContextTypes: {
+            t: PropTypes.func
+          }
+        }
+      )
+
+      expect(wrapper.find(Section)).toHaveLength(1)
+      expect(wrapper.find(Row)).toHaveLength(3)
+    })
+  })
+
+  describe('when no relevant data is defined', () => {
+    it('should not render', () => {
+      const wrapper = mount(<KeyInfosSection account={{}} />, {
+        context: {
+          t: key => key
+        },
+        childContextTypes: {
+          t: PropTypes.func
+        }
+      })
+
+      expect(wrapper.html()).toBe('')
+    })
+  })
+})
+
+describe('payments section', () => {
+  describe('when at least one relevant data is defined', () => {
+    it('should render defined data', () => {
+      const wrapper = mount(
+        <PaymentsSection
+          account={{
+            loan: {
+              lastPaymentAmount: 1000,
+              nextPaymentAmount: 1000
+            }
+          }}
+        />,
+        {
+          context: {
+            t: key => key
+          },
+          childContextTypes: {
+            t: PropTypes.func
+          }
+        }
+      )
```

Comment:
Instead of manually mocking the context, I thin using the I18n component passing a fake t is less brittle